### PR TITLE
Make links in text stylable via the MarkdownTypography class

### PR DESCRIPTION
**Pull request by aegis123**
make link text spanStyle configurable via the `MarkdownTypography` class this allows use to change it if they want change the fontWeight or text decoration.

### CODE REVIEW

File: multiplatform-markdown-renderer/src/commonMain/kotlin/com/mikepenz/markdown/utils/AnnotatedStringKtx.kt [hunk 1]
```diff
@@ -51,12 +48,9 @@ internal fun AnnotatedString.Builder.appendAutoLink(content: String, node: ASTNo
     } ?: node
     val destination = targetNode.getTextInNode(content).toString()
     pushStringAnnotation(MARKDOWN_TAG_URL, (destination))
+    LocalMarkdownTypography.current.link.toSpanStyle()
```
Comment (mikepenz) [inline multiplatform-markdown-renderer/src/commonMain/kotlin/com/mikepenz/markdown/utils/AnnotatedStringKtx.kt:51]: Is this line a left-over?

File: multiplatform-markdown-renderer-m2/src/commonMain/kotlin/com/mikepenz/markdown/m2/MarkdownTypography.kt [hunk 2]
```diff
@@ -23,9 +26,14 @@ fun markdownTypography(
     paragraph: TextStyle = MaterialTheme.typography.body1,
     ordered: TextStyle = MaterialTheme.typography.body1,
     bullet: TextStyle = MaterialTheme.typography.body1,
-    list: TextStyle = MaterialTheme.typography.body1
+    list: TextStyle = MaterialTheme.typography.body1,
+    link: TextStyle = MaterialTheme.typography.body1.copy(
+        fontWeight = FontWeight.Bold,
+        color = markdownColor().linkText,
```
Comment (mikepenz) [inline multiplatform-markdown-renderer-m2/src/commonMain/kotlin/com/mikepenz/markdown/m2/MarkdownTypography.kt:32]: This is potentially a different color from what `LocalMarkdownColors.current.linkText` ultimately holds.

Comment (mikepenz) [reply multiplatform-markdown-renderer-m2/src/commonMain/kotlin/com/mikepenz/markdown/m2/MarkdownTypography.kt:32]: Please remove the color here, and apply it in place where the link is used, as it was defined before.

File: multiplatform-markdown-renderer/src/commonMain/kotlin/com/mikepenz/markdown/utils/AnnotatedStringKtx.kt (hunk 1 above)
Comment (aegis123) [reply multiplatform-markdown-renderer/src/commonMain/kotlin/com/mikepenz/markdown/utils/AnnotatedStringKtx.kt]: yes that was a left-over, removed it.

File: multiplatform-markdown-renderer-m2/src/commonMain/kotlin/com/mikepenz/markdown/m2/MarkdownTypography.kt (hunk 2 above)
Comment (aegis123) [reply multiplatform-markdown-renderer-m2/src/commonMain/kotlin/com/mikepenz/markdown/m2/MarkdownTypography.kt]: removed the color in m2 and m3 typography creation and moved it the `AnnotatedString` extensions

File: multiplatform-markdown-renderer-m2/src/commonMain/kotlin/com/mikepenz/markdown/m2/MarkdownTypography.kt [hunk 3]
```diff
@@ -6,6 +6,9 @@ import androidx.compose.ui.text.SpanStyle
 import androidx.compose.ui.text.TextStyle
 import androidx.compose.ui.text.font.FontFamily
 import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextDecoration
+import com.mikepenz.markdown.compose.LocalMarkdownColors
```
Comment (mikepenz) [inline multiplatform-markdown-renderer-m2/src/commonMain/kotlin/com/mikepenz/markdown/m2/MarkdownTypography.kt:11]: can you please cleanup unused imports again

File: multiplatform-markdown-renderer-m3/src/commonMain/kotlin/com/mikepenz/markdown/m3/MarkdownTypography.kt [hunk 4]
```diff
@@ -6,6 +6,9 @@ import androidx.compose.ui.text.SpanStyle
 import androidx.compose.ui.text.TextStyle
 import androidx.compose.ui.text.font.FontFamily
 import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextDecoration
+import com.mikepenz.markdown.compose.LocalMarkdownColors
```
Comment (mikepenz) [inline multiplatform-markdown-renderer-m3/src/commonMain/kotlin/com/mikepenz/markdown/m3/MarkdownTypography.kt:11]: please cleanup unused import

File: multiplatform-markdown-renderer/src/commonMain/kotlin/com/mikepenz/markdown/model/MarkdownTypography.kt [hunk 5]
```diff
@@ -1,6 +1,7 @@
 package com.mikepenz.markdown.model
 
 import androidx.compose.runtime.Immutable
+import androidx.compose.ui.text.SpanStyle
```
Comment (mikepenz) [inline multiplatform-markdown-renderer/src/commonMain/kotlin/com/mikepenz/markdown/model/MarkdownTypography.kt:4]: please cleanup unused import

**Review by mikepenz — APPROVED** [review]

**Pull request merged**